zebra-network/src/protocol/external/tests/preallocate.rs

```diff
@@ -58,3 +61,60 @@ proptest! {
         assert!(largest_allowed_serialized.len() <= MAX_PROTOCOL_MESSAGE_LEN);
     }
 }
+
+proptest! {
+    /// Confirm that each MetaAddr takes exactly META_ADDR_SIZE bytes when serialized.
+    /// This verifies that our calculated `TrustedPreallocate::max_allocation()` is indeed an upper bound.
+    #[test]
+    fn meta_addr_size_is_correct(addr in MetaAddr::arbitrary()) {
+        zebra_test::init();
+
+        // We require sanitization before serialization
+        let addr = addr.sanitize();
+        prop_assume!(addr.is_some());
+        let addr = addr.unwrap();
+
+        let serialized = addr
+            .zcash_serialize_to_vec()
+            .expect("Serialization to vec must succeed");
+        assert!(serialized.len() == META_ADDR_SIZE)
+    }
+
+    /// Verifies that...
+    /// 1. The smallest disallowed vector of `MetaAddrs`s is too large to fit in a legal Zcash message
+    /// 2. The largest allowed vector is small enough to fit in a legal Zcash message
+    #[test]
+    fn meta_addr_max_allocation_is_correct(addr in MetaAddr::arbitrary()) {
+        zebra_test::init();
+
+        // We require sanitization before serialization
+        let addr = addr.sanitize();
+        prop_assume!(addr.is_some());
+        let addr = addr.unwrap();
+
+        let max_allocation: usize = MetaAddr::max_allocation().try_into().unwrap();
+        let mut smallest_disallowed_vec = Vec::with_capacity(max_allocation + 1);
+        for _ in 0..(MetaAddr::max_allocation() + 1) {
+            smallest_disallowed_vec.push(addr);
+        }
```

Comment:
Optional suggestion:

```suggestion
        let mut smallest_disallowed_vec = vec![addr; max_allocation + 1];
```

Reply:
I think I'll fix this on top of PR #3022, because it can be fixed in all the preallocate tests.
(And I need to write new preallocate tests in #3022.)